### ISSUE TITLE
Why is SplineConv slower to train even on GPU?

### ISSUE THREAD

**Issue by arshishir** [rusty1s/pytorch_geometric#1425]
## ❓ Questions & Help

I have found `SplineConv` to be much slower to train than to train `ChebConv` and `GCNConv`. For a task I am trying to accomplish, I want the GNN model that captures anisotropy. This means I am limited to `GMMConv` and `SplineConv` (am I missing some other models?). But these models take a lot of time to train. With `GMMConv` with `separate_gaussians=True`, I run out of GPU memory. It seems like `SplineConv` captures anisotropy better than `GMMConv`. I do not know why this is the case though. `GMMConv` is much general model than `SplineConv` and `GMMConv` is designed to capture anisotropy, right?

Has anyone encountered these issues? Any suggestions would be extremely helpful.

**Comment by rusty1s** [comment]
Hi, thanks for your interest. The main reason those models are slower is because they utilize different weight matrices for different edges, resulting in a higher memory footprint and overall more complex implementation.

The main reason that I see why `SplineConv` outperforms `GMMConv` in most experiments is that I feel it provides overall better gradients during training, while `GMMConv` seems to rely much heavier on a better initialization. You are right that `GMMConv`, in theory, should be a bit better to capture anisotropy, but IMO this does not transfer to training, where I feel it gets stuck in bad local minima.

As a third alternative, you can also look at `NNConv`, which should be the most general one out of the three.

**Comment by arshishir** [comment]
Thanks @rusty1s . I am trying to understand how the sense of direction is encoded in `SplineConv`. 

For example in  MNIST, where the underlying graph is a grid graph. The graph used is bidirectional with self-connection. This leads to the total number of 6724 edges. The edge feature is defined as some kind of relative coordinate/ pseudo coordinate. What is the precise definition of pseudo coordinate? Is there an intuitive way to understand pseudo coordinates? How should I determine if I should choose Cartesian or polar? In MoNet paper, they have used `Polar` for MNIST and in `SplineConv` paper, you guys have used `Cartesian`. Was this choice arbitrary or was it based on some insight?

 `SplineConv` should be equivalent to `CNN` for MNIST, right? If so then, we should be able to get the same level of accuracy as that of `CNN` for the majority of tasks where `CNN` has been successful, right?
